### ISSUE TITLE
Custom color scheme for choropleth

### ISSUE THREAD

**Issue by lecardozo** [python-visualization/folium#403]
Hi guys,
Thanks for the nice work here!

I'm new here so I'm sorry if I'm missing something. Do we have, in the current version, any way to define specific color schemes for the choropleth, beyond those defined in the docsting of the function (BuGn, BuPu, GnBu, OrRd, PuBu, PuBuGn, PuRd, RdPu, YlGn, YlGnBu, YlOrBr, and YlOrRd)?

**Comment by eddies** [comment]
Assuming you're using v0.2.0 and the `choropleth` convenience method, you are indeed limited to the `color_brewer` schemes defined in https://github.com/python-visualization/folium/blob/v0.2.0/folium/utilities.py#L104

On the other hand, for maximum flexibility, if you create your choropleth via either the GeoJson or TopoJson classes, you can pass in your own instance of `colormap`.

For examples, have a look at the StepColormap and LinearColormap sections in http://nbviewer.jupyter.org/github/python-visualization/folium/blob/v0.2.0/examples/Colormaps.ipynb

**Comment by ocefpaf** [comment]
Thanks @eddies!

**Comment by lecardozo** [comment]
Nice! Thanks @eddies!

**Comment by FakieKickflip** [comment]
It says that there is no module named colormap in folium.

**Comment by Conengmo** [comment]
Please see the updated notebook: https://nbviewer.jupyter.org/github/python-visualization/folium/blob/master/examples/Colormaps.ipynb

There are two ways to import the colormap classes:
- From their source module: `branca.colormap import ColorMap, LinearColormap, StepColormap`
- From a shortcut: `from folium import ColorMap, LinearColormap, StepColormap`

**Comment by FakieKickflip** [comment]
Thank you! 

I have another question: I have a folium.Choropleth and I want to pass a customised map with 10 different colors into the fill_color parameter. I am using a df that has values between 1 and 10 in one column and if there is for example a 2 it should be shown blue, for a 7 yellow and so on...

**Comment by lidiasm** [comment]
Hi!

I'd like to do the same thing, did you find a way?

**Comment by leonardas103** [comment]
TLDR
Create your  [colourmap](https://nbviewer.jupyter.org/github/python-visualization/folium/blob/master/examples/Colormaps.ipynb), and inject it:
```
choropleth.geojson.style_function = lambda x: {...}
choropleth.add_to(m)
```
---
Long version: Let's say you have a df with columns (zip_codes, count):

Variables
```
df = pd.read_csv(...)
m = folium.Map(...)
choropleth = folium.Choropleth(...)
````

Create your desired [colourmap](https://nbviewer.jupyter.org/github/python-visualization/folium/blob/master/examples/Colormaps.ipynb), using max and min from your data range:
```
colourstep = folium.LinearColormap(['green','yellow','red'], vmin=3., vmax=10.).to_step(4)
```

Create a dict from the df:
```
df_dict = df.set_index('zip_code')['count']
```

Replace the colour style
```
style_f = lambda x: {'fillColor': colourstep(df_dict[x['properties']['postalCode']]), "color": "black", "weight": 2}
choropleth.geojson.style_function = style_f
choropleth.add_to(m)
```
Note that my geojson zone ids were found at `geojson['features'][0]['properties']['postalCode']`, so `key_on='feature.properties.postalCode'` likewise this became `x['properties']['postalCode']` in the style function.   
Lastly, your will need to fix your legend by altering [`choropleth.colour_scale`](https://github.com/python-visualization/folium/blob/c25c84c888415a8f242e133c97242e26c3070fe6/folium/features.py#L1229])

Remarks
---
It is easier to just use one of the defined colours, reversing the colours order with `_r` e.g `fill_color='RdYlGn_r'`
ColorBrewer codes for [reference](https://github.com/python-visualization/branca/blob/master/branca/scheme_base_codes.json):
![image](https://user-images.githubusercontent.com/17128994/115975254-c2031e00-a56b-11eb-8025-d82d36bfda1d.png)
